katdal/applycal.py

```diff
@@ -406,11 +435,18 @@ def calc_correction(chunks, cache, corrprods, cal_products,
         Selected correlation products as pairs of correlator input labels
     cal_products : sequence of string
         Calibration products that will contribute to corrections (e.g. ["l1.G"])
+    data_freqs : array of float, shape (*F*,)
+        Centre frequency of each frequency channel of visibilities, in Hz
+    all_cal_freqs : dict
+        Dictionary mapping cal stream name to array of associated frequencies
```

Comment:
The synthesised stream name. The underlying stream names only appear in the call to `add_applycal_sensors` and everything else uses "the" name.

Reply:
It would be nice to have that mentioned somewhere in the documentation. I found it pretty confusing during the review to know which one should be used where. But I'm happy if you fold it into the next PR rather than trying to wedge it in here.

Reply:
If we want to drive these decisions from telstate then we need to get it figured out ASAP. If we leave it until L2 is "done" and we start making using it, it'll be difficult to change because we'll have files in the archive without the necessary telstate keys.

Reply:
I added an example to hopefully clear it up.